odoo/sql_db.py

```diff
@@ -221,6 +268,15 @@ def execute(self, query, params=None, log_exceptions=None):
             _logger.debug("query: %s", self._obj.mogrify(query, params).decode(encoding, 'replace'))
         now = time.time()
         try:
+            if re.match(r'^\s*select', query, re.IGNORECASE):
```

Comment:
I think we need a more specific regex, because, or a way to change it at discretion

Reply:
That is a future feature that I have thought about the main thing here is to validate that this concept is the proper one, regex can be done through a param too or an ENV variable maybe.